src/respend/respendrelayer.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2018 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "respend/respendrelayer.h"
+#include "net.h" // RelayTransaction
+#include "options.h"
+#include "primitives/transaction.h"
+#include "protocol.h"
+#include "streams.h"
+#include "util.h"
+#include "utilprocessmsg.h"
+#include <mutex>
+
+namespace respend {
+
+namespace {
+
+// Apply an independent rate limit to double-spend relays
+class RelayLimiter {
+    public:
+        RelayLimiter() : respendCount(0), lastRespendTime(0) { }
+
+        bool HasLimitExceeded(const CTransaction& doubleSpend)
+        {
+            unsigned int size = ::GetSerializeSize(doubleSpend,
+                                                   SER_NETWORK, PROTOCOL_VERSION);
+
+            std::lock_guard<std::mutex> lock(cs);
+            int64_t limit = Opt().RespendRelayLimit();
+            if (RateLimitExceeded(respendCount, lastRespendTime, limit, size)) {
+                LogPrint("respend", "respend: Double-spend relay rejected by rate limiter\n");
+                return true;
+            }
+
+            LogPrint("respend", "respend: Double-spend relay rate limited: %g => %g\n",
```

Comment:
Tiny change request: `limited` -> `limiter`. The limiter is just reporting the accumulation, not an event where the rate was limited.